os/board/esp32_devkit/src/esp32_boot.c

```diff
@@ -234,6 +237,9 @@ void board_initialize(void)
 {
 	/* Perform board-specific initialization */
 	(void)esp32_bringup();
+#ifdef CONFIG_ESP32_SPIFLASH
+	esp_partition_initialize();
```

Comment:
Same as above, it should be done after spi initialize but I think it has only 1 spi port, so it doesn't initialize spi explicitly.
so instead of this, you can use up_flashinitialize in esp32_flash.c, I think.
This assume that never access flash directly before MTD driver initialize.